go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1936,6 +1960,52 @@ func commonFlows(bridge *bridgeConfiguration) ([]string, error) {
 						fmt.Sprintf("cookie=%s, priority=100, in_port=%s, dl_src=%s, ipv6, "+
 							"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:%s",
 							defaultOpenFlowCookie, netConfig.ofPortPatch, bridgeMacAddress, config.Default.ConntrackZone, netConfig.masqCTMark, ofPortPhys))
+
+					// IP traffic from the OVN network to the host network should be handled normally by the bridge instead of
+					// being forwarded directly to the NIC (prio=100 flow above). This allows pods in the default network to reach
+					// localnet pods on the same node.
+					if config.Gateway.Mode == config.GatewayModeShared || config.Gateway.Mode == config.GatewayModeLocal {
+						for _, hostSubnet := range hostSubnets {
+							if hostSubnet.IP.To4() != nil {
+								continue
+							}
+							inPort := ""
+							ctMark := ""
+							if config.Gateway.Mode == config.GatewayModeShared {
+								inPort = netConfig.ofPortPatch
+								ctMark = netConfig.masqCTMark
+							} else if config.Gateway.Mode == config.GatewayModeLocal {
+								inPort = "LOCAL"
+								ctMark = ctMarkHost
+							}
```

Comment:
We are using this at ipv4 too, let's encpasulate on function or create a map to define inport and ctmark per gateway mode (also I think this kind of mapping is useful at other part of the code here).

Reply:
yes, I captured that in a separate function in the end, as suggested by Miguel